plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/github.ts

```diff
@@ -263,7 +263,22 @@ export function createPublishGithubAction(options: {
               allow_rebase_merge: allowRebaseMerge,
             });
 
-      const { data: newRepo } = await repoCreationPromise;
+      let newRepo;
+
+      try {
+        newRepo = (await repoCreationPromise).data;
+      } catch (e) {
+        assertError(e);
+        if (e.message === 'Resource not accessible by integration') {
+          ctx.logger.warn(
+            'The GitHub app or token provided to Backstage may not have the required permissions to create the repository.',
```

Comment:
Generally agains logging and throwing, but seems alright in this case. Should we not include the owner/repo that was attempted in this message too though?